### PR TITLE
fix: swaps test and others

### LINKED ISSUE #4725: Fix testnet swaps tests
When we hid the swaps button on testnet, the swaps tests broke.

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of Leather — [Extension build](https://github.com/leather-wallet/extension/actions/runs/7256172772), [Test report](https://leather-wallet.github.io/playwright-reports/fix/swaps-test)

This PR changes to test swaps on mainnet bc we hide swaps on testnet. The broadcast is caught and aborted, so it should be ok. I also just did some clean up of the locators and handled if there happens to be a pending tx (which means we needed to catch the Hiro api post route).

### CODE REVIEW

File: tests/specs/rpc-stacks-transaction/transaction-signing.spec.ts
Comment (fbwoolf) [inline tests/specs/rpc-stacks-transaction/transaction-signing.spec.ts:56]: This test was failing bc the high fee warning popup was being triggered w/out a fee provided.

File: tests/specs/transactions/transactions.spec.ts
Comment (fbwoolf) [inline tests/specs/transactions/transactions.spec.ts:28]: Account 1 in the test account doesn't have a 0 balance atm so this was failing.

**Comment by fbwoolf** [comment]
Fixed a few other failing tests here.

File: tests/specs/swap/swap.spec.ts
Comment (kyranjamie) [inline tests/specs/swap/swap.spec.ts:4]: also possible to be specific to Hiro api origin, too?

File: tests/specs/transactions/transactions.spec.ts
Comment (kyranjamie) [reply tests/specs/transactions/transactions.spec.ts:28]: Should we fill it up instead of making this change?

**Review by kyranjamie — APPROVED** [review]
Great investigation @fbwoolf, thanks

File: tests/specs/transactions/transactions.spec.ts
Comment (pete-watters) [reply tests/specs/transactions/transactions.spec.ts:28]: I think the problem was `Account 1` has funds so clicking on `Account 2` which doesn't solves it

**Review by pete-watters — APPROVED** [review]
Nice work @fbwoolf 👍 

I think we need to look into how we can better block merge of PRs with failing tests to stop this happening. It seems [this PR](https://github.com/leather-wallet/extension/pull/4698) introduced these fails

I see some suggestions from Kyran that could be good to add but I will merge this now and we can iterate on it if needed

**Pull request merged**
